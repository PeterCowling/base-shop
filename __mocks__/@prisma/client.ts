import { jest } from '@jest/globals';

const editChanges: any[] = [];

class PrismaClient {
  editChange = {
    findMany: jest.fn(async () => [...editChanges]),
    count: jest.fn(async () => editChanges.length),
  };
}

export { PrismaClient, editChanges as __editChanges };
