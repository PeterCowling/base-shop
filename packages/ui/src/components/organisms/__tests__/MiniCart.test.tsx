import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MiniCart } from "../MiniCart.client";

jest.mock("@platform-core/src/contexts/CurrencyContext", () => ({
  useCurrency: () => ["USD", jest.fn()],
}));

const mockUseCart = jest.fn();
jest.mock("@platform-core/src/contexts/CartContext", () => ({
  useCart: () => mockUseCart(),
}));

describe("MiniCart", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders empty state when cart has no items", async () => {
    mockUseCart.mockReturnValue([{}, jest.fn()]);
    render(<MiniCart trigger={<button>Open Cart</button>} />);

    await userEvent.click(screen.getByText("Open Cart"));
    expect(await screen.findByText("Cart is empty.")).toBeInTheDocument();
  });

  it("shows cart items and handles quantity changes and removal", async () => {
    const dispatch = jest.fn();
    mockUseCart.mockReturnValue([
      {
        "sku1:m": {
          sku: { id: "sku1", title: "Item", price: 10, sizes: [] },
          qty: 1,
          size: "m",
        },
      },
      dispatch,
    ]);

    const { baseElement } = render(<MiniCart trigger={<button>Cart</button>} />);
    await userEvent.click(screen.getByText("Cart"));
    expect(await screen.findByText("Item")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();

    await userEvent.click(
      screen.getByRole("button", { name: /increase quantity/i })
    );
    expect(dispatch).toHaveBeenCalledWith({
      type: "setQty",
      id: "sku1:m",
      qty: 2,
    });

    await userEvent.click(
      screen.getByRole("button", { name: /decrease quantity/i })
    );
    expect(dispatch).toHaveBeenCalledWith({
      type: "setQty",
      id: "sku1:m",
      qty: 0,
    });

    await userEvent.click(screen.getByRole("button", { name: /remove/i }));
    expect(dispatch).toHaveBeenCalledWith({ type: "remove", id: "sku1:m" });
  });
});
