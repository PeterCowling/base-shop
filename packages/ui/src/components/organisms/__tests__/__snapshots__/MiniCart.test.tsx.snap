// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCart shows cart items and handles quantity changes and removal 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-«r3»"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Cart
    </button>
  </div>
  <div
    aria-hidden="true"
    class="data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out data-[state=open]:fade-in fixed inset-0 z-50 bg-fg/50 backdrop-blur-sm"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto;"
  />
  <div
    aria-describedby="radix-«r5»"
    aria-labelledby="radix-«r4»"
    class="bg-background data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 fixed top-[50%] left-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border p-6 shadow-lg duration-200 bg-background fixed top-0 right-0 h-full max-w-full rounded-none border-l p-6 shadow-lg w-80"
    data-state="open"
    id="radix-«r3»"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <h2
      class="text-lg font-semibold mb-4"
      id="radix-«r4»"
    >
      Your Cart
    </h2>
    <p
      class="text-muted-foreground text-sm sr-only"
      id="radix-«r5»"
    >
      Review items in your cart
    </p>
    <div
      class="flex h-full flex-col gap-4"
    >
      <ul
        class="grow space-y-3 overflow-y-auto"
      >
        <li
          class="flex items-center justify-between gap-2"
        >
          <span
            class="text-sm"
          >
            Item
            <span
              class="ml-1 text-muted"
            >
              (
              m
              )
            </span>
          </span>
          <div
            class="flex items-center gap-1"
          >
            <button
              aria-label="Decrease quantity"
              class="inline-flex items-center justify-center rounded-md text-sm font-medium text-fg transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-fg hover:bg-primary/90 px-2 py-1 text-xs"
              data-token="--color-primary"
            >
              -
            </button>
            <span
              class="text-sm"
            >
              1
            </span>
            <button
              aria-label="Increase quantity"
              class="inline-flex items-center justify-center rounded-md text-sm font-medium text-fg transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-fg hover:bg-primary/90 px-2 py-1 text-xs"
              data-token="--color-primary"
            >
              +
            </button>
          </div>
          <button
            class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 px-4 py-2 bg-destructive text-destructive-foreground hover:bg-destructive/90 px-2 py-1 text-xs"
          >
            Remove
          </button>
        </li>
      </ul>
      <div
        class="border-t pt-4 text-sm"
      >
        <div
          class="flex justify-between font-semibold"
        >
          <span>
            Subtotal
          </span>
          <span
            class=""
          >
            $10.00
          </span>
        </div>
      </div>
    </div>
    <button
      class="ring-offset-background focus:ring-ring data-[state=open]:bg-secondary absolute top-4 right-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:ring-2 focus:ring-offset-2 focus:outline-none disabled:pointer-events-none"
      type="button"
    >
      <svg
        class="h-4 w-4"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="sr-only"
      >
        Close
      </span>
    </button>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
